src/Sulu/Bundle/MediaBundle/Media/ImageConverter/ImagineImageConverter.php

```diff
@@ -12,116 +12,187 @@
+     *
+     * @param ImageInterface $image
+     * @param $tansformations
      *
      * @throws ImageProxyInvalidFormatOptionsException
+     *
+     * @return ImageInterface The modified image
+     */
+    private function applyTransformations(ImageInterface $image, $tansformations)
+    {
+        foreach ($tansformations as $transformation) {
+            if (!isset($transformation['effect'])) {
+                throw new ImageProxyInvalidFormatOptionsException('Effect not found');
+            }
+            $image = $this->transformAllLayers(
+                $image,
+                function (ImageInterface &$layer) use ($transformation) {
+                    return $this->transformationPool->get($transformation['effect'])->execute(
+                        $layer,
+                        $transformation['parameters']
+                    );
+                }
+            );
+        }
+
+        return $image;
+    }
+
+    /**
+     * Scales a given image according to the information passed as the second argument.
+     *
+     * @param ImageInterface $image
+     * @param $scale
+     *
+     * @return ImageInterface
      */
-    public function call($command, $parameters)
+    private function applyScale(ImageInterface $image, $scale)
     {
-        if (count($this->image->layers())) {
-            $counter = 0;
-            $this->image->layers()->coalesce();
-            foreach ($this->image->layers() as $layer) {
-                ++$counter;
-                $this->commandManager->get($command)->execute($layer, $parameters);
-                if ($counter == 1) {
-                    /** @var \Imagine\Imagick\Image|\Imagine\Gd\Image $image */
-                    $image = $layer; // use first layer as main image
+        return $this->transformAllLayers(
+            $image,
+            function (ImageInterface &$layer) use ($scale) {
+                return $this->scaling->scale(
+                    $layer,
+                    $scale['x'],
+                    $scale['y'],
+                    $scale['mode'],
+                    $scale['forceRatio'],
+                    $scale['retina']
+                );
+            }
+        );
+    }
+
+    /**
+     * Ensures that the color mode of the passed image is RGB.
+     *
+     * @param ImageInterface $image
+     *
+     * @return ImageInterface $image The modified image
+     */
+    private function toRGB(ImageInterface $image)
+    {
+        if ($image->palette()->name() == 'cmyk') {
+            $image->usePalette(new RGB());
+        }
+
+        return $image;
+    }
+
+    /**
+     * Applies a callback to every layer of an image and returns the resulting image.
+     *
+     * @param ImageInterface $image
+     * @param \Closure $closure The closure to apply to all layers
+     *
+     * @return ImageInterface
+     */
+    private function transformAllLayers(ImageInterface $image, \Closure $closure)
```

Comment:
I made the parameter of type `callable` but renamed it to `$callable` as we use "transformation" for another concept, the transformations which can be configured in the XML. As these transformations also appear in this file this could lead to confusion.